koku/api/settings/test/tag_mappings/test_view.py

```diff
@@ -0,0 +1,288 @@
+            response = self.client.put(url, data, format="json", **self.headers)
+
+            self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_put_method_validate_child(self):
+        """Test if a child can be added as a parent."""
+
+        # Adding sample uuids
+        random_uuid_list = self.retrieve_sample_uuids()
+        url = reverse("tags-mapping-child-add")
+        data = {
+            "parent": random_uuid_list[0],
+            "children": [random_uuid_list[1], random_uuid_list[2], random_uuid_list[3]],
+        }
+        response = self.client.put(url, data, format="json", **self.headers)
+
+        if response.status_code == status.HTTP_200_OK:
+            # Adding a child as parent
+            data = {"parent": random_uuid_list[2], "children": [random_uuid_list[4]]}
+            response = self.client.put(url, data, format="json", **self.headers)
+
+            self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_put_method_add_multiple_children(self):
+        """Test adding multiple children (list)."""
+        random_uuid_list = self.retrieve_sample_uuids()
+        url = reverse("tags-mapping-child-add")
+        data = {"parent": random_uuid_list[0], "children": [random_uuid_list[1], random_uuid_list[2]]}
+
+        response = self.client.put(url, data, format="json", **self.headers)
+        self.assertEqual(response.status_code, status.HTTP_203_NON_AUTHORITATIVE_INFORMATION)
+
+    def test_put_method_remove_children(self):
+        """Test removing children."""
+
+        # Adding sample uuids
+        random_uuid_list = self.retrieve_sample_uuids()
+        url = reverse("tags-mapping-child-add")
+        data = {
+            "parent": random_uuid_list[0],
+            "children": [random_uuid_list[1], random_uuid_list[2], random_uuid_list[3]],
+        }
+        response = self.client.put(url, data, format="json", **self.headers)
+
+        if response.status_code == status.HTTP_203_NON_AUTHORITATIVE_INFORMATION:
+            # Removing children
+            url = reverse("tags-mapping-child-remove")
+            data = {"ids": [random_uuid_list[1], random_uuid_list[3]]}
+            response = self.client.put(url, data, format="json", **self.headers)
+
+            self.assertEqual(response.status_code, status.HTTP_203_NON_AUTHORITATIVE_INFORMATION)
+
+    def test_put_method_remove_parent(self):
+        """Test removing parent."""
+
+        # Adding sample uuids
+        random_uuid_list = self.retrieve_sample_uuids()
+        url = reverse("tags-mapping-child-add")
+        data = {
+            "parent": random_uuid_list[0],
+            "children": [random_uuid_list[1], random_uuid_list[2], random_uuid_list[3]],
+        }
+        response = self.client.put(url, data, format="json", **self.headers)
+
+        if response.status_code == status.HTTP_203_NON_AUTHORITATIVE_INFORMATION:
```

Comment:
This test should be updated to remove this if statement. The above should be written so that the response is always the correct status code which ensures the code below is run.